components/raftstore/src/store/fsm/peer.rs

```diff
@@ -3163,19 +3163,20 @@ where
             "region" => ?region,
         );
 
-        let mut state = self.ctx.global_replication_state.lock().unwrap();
-        let gb = state
-            .calculate_commit_group(self.fsm.peer.replication_mode_version, region.get_peers());
-        self.fsm.peer.raft_group.raft.clear_commit_group();
-        self.fsm.peer.raft_group.raft.assign_commit_groups(gb);
-
         let mut meta = self.ctx.store_meta.lock().unwrap();
         debug!(
             "check snapshot range";
             "region_id" => self.fsm.region_id(),
             "peer_id" => self.fsm.peer_id(),
             "prev_region" => ?prev_region,
         );
+        {
+            let mut state = self.ctx.global_replication_state.lock().unwrap();
+            let gb = state
+                .calculate_commit_group(self.fsm.peer.replication_mode_version, region.get_peers());
+            self.fsm.peer.raft_group.raft.clear_commit_group();
+            self.fsm.peer.raft_group.raft.assign_commit_groups(gb);
+        }
```

Comment:
If it's no need to refer to `store_meta`, I think it's better to move this code beyond line 3166.
If the `global_replication_state lock` is released before acquiring `store_meta lock`, there is no deadlock.

Reply:
Furthermore, how about adding some comments for `global_replication_state lock` like `pending_create_peers`
https://github.com/tikv/tikv/blob/master/components/raftstore/src/store/fsm/store.rs#L370-L375.